File: Deepspace-Ruby/lib/ShieldBooster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'ShieldToUI'
require_relative 'CombatElement'


module Deepspace
class ShieldBooster
  include CombatElement
  attr_reader :name
  
  def initialize(n, b, u)
    @name=n
    @boost=b
    @uses=u
  end
  
  
  
  def self.newCopy(s)
    nombre=s.name
    boostt=s.boost
    usess=s.uses
    ShieldBooster.new(s.name, s.boost, s.uses)
  end
  
  def getUIversion
    ShieldToUI.new(self)
  end
  
  public
  def boost
    @boost
  end
  
  public
  def uses
    @uses
  end
  
  public
  def useIt
    if (@uses > 0)
      @uses=@uses-1
      boost
    else 
      resultado=1.0
      resultado
    end
  end
  

  
  
  
  def to_s
    return "ShieldBooster: name=#{@name}, boost=#{@boost}, uses=#{@uses}"
  end
end
end

File: Deepspace-Java/src/deepspace/ShieldBooster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author inaki
 */
public class ShieldBooster implements CombatElement{
    private String name;
    private float boost;
    private int uses;
    
    ShieldBooster(String n, float b, int u){
        this.name=n;
        this.boost=b;
        this.uses=u;
    }
    
    ShieldBooster(ShieldBooster s){
        this.name=s.name;
        this.boost=s.boost;
        this.uses=s.uses;
    }
    
    ShieldToUI getUIversion (){
        return new ShieldToUI(this);
    }
    
    public float getBoost(){
        return this.boost;
    }
    
    @Override
    public int getUses(){
        return this.uses;
    }
    
    @Override
    public float useIt(){
        if (this.uses > 0){
            uses--;
            return boost;
        }
        else {
            float resultado=1;
            return resultado;
        }
    }

    @Override
    public String toString() {
        return "ShieldBooster{" + "name=" + name + ", boost=" + boost + ", uses=" + uses + '}';
    }
    

}
